src/main/java/frc/robot/subsystems/ShooterPivot.java

```diff
@@ -97,12 +100,40 @@ private void setPosition(Rotation2d position) {
     setpoint = position.getRadians();
   }
 
+  public Rotation2d getEstimatedAngle(double distance) {
+    double angle = Constants.ShooterPivot.a * distance + Constants.ShooterPivot.b;
+    return Rotation2d.fromDegrees(angle);
+  }
+
+  public double speakerDistance(Pose2d RobotPose) {
+    if (DriverStation.getAlliance().isPresent()) {
+      if (DriverStation.getAlliance().get() == DriverStation.Alliance.Red) {
+        double redSpeakerDistance =
+            Math.hypot((RobotPose.getX() - 16.58), (RobotPose.getY() - 5.55));
+        return redSpeakerDistance;
+
+      } else {
+        double blueSpeakerDistance = Math.hypot((RobotPose.getX() - 0), (RobotPose.getY() - 5.55));
+        return blueSpeakerDistance;
+      }
+    }
+    return 0.0;
+  }
+
+  // to do make this part of the shooting process
   public Command setPositionCommand(Rotation2d position) {
     return new InstantCommand(() -> this.setPosition(position), this);
   }
 
-  public double linnerInterpilation(double distance) {
-    double angle = Constants.ShooterPivot.a * distance + Constants.ShooterPivot.b;
-    return angle;
+  public void setRestPosition(Rotation2d position) {
```

Comment:
Redundant, just use setPosition

Reply:
the purpose of the method is to get the angle so we know which angle we are shooting at

Reply:
Agree with Jake that the implementation here is redundant with setPosition.

General points about a pivot arm.
When the robot starts up, we can assume that the starting configuration is arm at rest. We can zero the absolute encoder when the shooter is constructed. 
If the system slips during use, we will need an additional method for the driver to rezero during operation.

When the arm is in its rest position, does the motor require power to hold it in place? For that matter, after use, does it need to be lowered under power or can it be turned off and allowed to coast down? Brake mode?